src/context/online.rs

```diff
@@ -140,15 +165,73 @@ impl AudioContext {
     /// the time at which the first sample in the buffer is actually processed
     /// by the audio output device.
     #[must_use]
+    #[allow(clippy::missing_panics_doc)]
     pub fn output_latency(&self) -> f64 {
-        self.backend.output_latency()
+        self.backend.lock().unwrap().output_latency()
     }
 
     /// Identifier or the information of the current audio output device.
     ///
     /// The initial value is `None`, which means the default audio output device.
-    pub fn sink_id(&self) -> Option<&str> {
-        self.backend.sink_id()
+    #[allow(clippy::missing_panics_doc)]
+    pub fn sink_id(&self) -> Option<String> {
+        self.backend.lock().unwrap().sink_id().map(str::to_string)
+    }
+
+    /// Update the current audio output device.
+    ///
+    /// This function operates synchronously and might block the current thread. An async version
+    /// is currently not implemented.
+    #[allow(clippy::needless_collect, clippy::missing_panics_doc)]
+    pub fn set_sink_id_sync(&self, sink_id: String) -> Result<(), Box<dyn Error>> {
```

Comment:
This is the meat of the PR.

1. check if sink is already active
2. validate sink
3. acquire necessary locks (avoid issues with concurrent calls)
4. shut down current render thread and recover the audio graph
5. boot up new thread, and ship the audio graph
6. release locks